src/main/java/org/junitpioneer/internal/PioneerUtils.java

```diff
@@ -82,6 +84,22 @@ private PioneerUtils() {
 		return method;
 	}
 
+	/**
+	 * Find all (parent) {@code ExtensionContext}s via {@link ExtensionContext#getParent()}.
+	 *
+	 * @param context the context for which to find all (parent) contexts; never {@code null}
+	 * @return a list of all contexts, including the given context; never {@code null} or empty
+	 */
+	public static List<ExtensionContext> findAllContexts(ExtensionContext context) {
+		List<ExtensionContext> contexts = new ArrayList<>();
+		Optional<ExtensionContext> currentContext = Optional.of(context);
+		do {
```

Comment:
I don't like `do...while` loops, but it looks super appropriate here. Dammit. ☹️

Reply:
Alternatively recursion?

```java
public static List<ExtensionContext> findAllContexts(ExtensionContext context) {
	List<ExtensionContext> allContexts = new ArrayList<>();
	allContexts.add(context);
	allContexts.addAll(context.getParent()
			.map(PioneerUtils::findAllContexts)
			.orElse(Collections.emptyList()));
	return allContexts;
}
```

Reply:
Very fancy alternative - in both ways (very elaborate and elegant). I actually prefer the loop, because it's simpler (if uglier). I was just complaining a little 😇